Dockerfile

```diff
@@ -8,9 +8,13 @@ RUN apt-get update \
         patch pkg-config swig unzip wget xz-utils \
         autoconf autotools-dev automake texinfo dejagnu \
         build-essential prelink autoconf libtool libltdl-dev \
-        gnupg2 libdbus-glib-1-2 sudo \
+        gnupg2 libdbus-glib-1-2 sudo curl \
   && rm -rf /var/lib/apt/lists/*
 
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+RUN source $HOME/.cargo/env
+RUN rustup target add wasm32-unknown-emscripten
```

Comment:
So normally you should be able to install it with [`make rust`](https://github.com/pyodide/pyodide/blob/main/Makefile#L242). Also it's better to group commands to avoid creating too many Docker layers (and `make rust` would take care of it).

Reply:
Thanks! Would this mean that I'd need to add `RUN git clone <pyodide repo url>` first so that `make` can access Pyodide's `Makefile`? And if so, where should the Pyodide repo be cloned to?

Oh unless you mean that we don't actually need this in the Dockerfile because users are expected to run `make` after running `./run_docker --pre-built`? ([related issue](https://github.com/pyodide/pyodide/issues/2705)) In that case, can the `rust` target be added as a default so that it's installed when people run `make`?

Reply:
Aww, you are right we don't have the source tree here. OK this is fine then.